src/main/java/io/jenkins/plugins/coverage/CoverageProcessor.java

```diff
@@ -143,6 +152,65 @@ private void convertResultToAction(CoverageResult coverageReport, HealthReport h
         }
     }
 
+    private void setDiffInCoverageForChangeRequest(CoverageResult coverageReport) {
+        // Use limited number of builds to don't search too long
+        final int numberOfBuildsFromTargetBranch = 50;
+        // Calculate diff only for line coverage
+        Ratio changeRequestLinesCoverage = coverageReport.getCoverage(CoverageElement.LINE);
+        if (changeRequestLinesCoverage != null) {
```

Comment:
Could you reduce the nesting level? There are too many nested if statement and hard to recognize.

Reply:
Agree, will try

Reply:
@cizezsy Please, check

Reply:
@tempora-mutantur Maybe we can use some codes like this
```
if (changeRequestLinesCoverage == null) {
    return
}

Job job = run.getParent();
parent = job.getParent();

if (!(parent instanceof MultiBranchProject)) {
    return
}

......
```

Your current approach will pass through all if statement even if the condition must be false.

Reply:
@cizezsy 
Thank you. Please, take a look, I've made such change